pandas/tests/generic/test_finalize.py

```diff
@@ -675,3 +669,154 @@ def test_finalize_frame_series_name():
     df = pd.DataFrame({"name": [1, 2]})
     result = pd.Series([1, 2]).__finalize__(df)
     assert result.name is None
+
+
+# ----------------------------------------------------------------------------
+# Tests for merge
+
+
+@pytest.mark.parametrize(
+    ["allow_on_left", "allow_on_right"],
+    [(False, False), (False, True), (True, False), (True, True)],
+)
+@pytest.mark.parametrize(
+    "how",
+    [
+        "left",
+        "right",
+        "inner",
+        "outer",
+        "left_anti",
+        "right_anti",
+        "cross",
+    ],
+)
+def test_merge_sets_duplication_allowance_flag(
+    how: MergeHow,
+    allow_on_left: bool,
+    allow_on_right: bool,
+):
+    """Check that DataFrame.merge correctly sets the allow_duplicate_labels flag
+    on its result.
+
+    The flag on the result should be set to true if and only if both arguments
+    to merge have their flags set to True.
+    """
```

Comment:
Can you remove docstrings from tests, we generally do not add these. Comments are fine if they are adding something that is not already in the code itself.